domain/src/test/java/org/oppia/android/domain/spotlight/SpotlightStateControllerTest.kt

```diff
@@ -0,0 +1,388 @@
+      spotlightStateController.retrieveSpotlightViewState(profileId, TOPIC_REVISION_TAB)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_promotedStories_notMarked_returnsSpotlightStateNotSeen() {
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, PROMOTED_STORIES)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_NOT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_promotedStories_marked_returnsSpotlightStateSeen() {
+    markSpotlightSeen(PROMOTED_STORIES)
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, PROMOTED_STORIES)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_lessonsBackButton_notMarked_returnsSpotlightStateNotSeen() {
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, LESSONS_BACK_BUTTON)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_NOT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_lessonsBackButton_marked_returnsSpotlightStateSeen() {
+    markSpotlightSeen(LESSONS_BACK_BUTTON)
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, LESSONS_BACK_BUTTON)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_voiceoverPlayIcon_notMarked_returnsSpotlightStateNotSeen() {
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, VOICEOVER_PLAY_ICON)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_NOT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_voiceoverPlayIcon_marked_returnsSpotlightStateSeen() {
+    markSpotlightSeen(VOICEOVER_PLAY_ICON)
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, VOICEOVER_PLAY_ICON)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_voiceoverLanguageIcon_notMarked_returnsSpotlightStateNotSeen() { // ktlint-disable max-line-length
```

Comment:
```suggestion
  fun testRetrieveSpotlightViewState_voiceoverLangIcon_notMarked_returnsSpotlightStateNotSeen() {
```

Prefer not having to disable the max-line length check. :)

Reply:
done